docs/machine-learning/tutorials/movie-recommmendation.md

```diff
@@ -0,0 +1,474 @@
+```csharp
+public static ITransformer BuildAndTrainModel(MLContext mlContext, IDataView trainingDataView)
+{
+
+
+}
+```
+> [!NOTE]
+> This method will give you an error until you add a return statement in the following steps.
+
+Define the data transformations by adding the following code to `BuildAndTrainModel()`:
+   
+[!code-csharp[DataTransformations](../../../samples/machine-learning/tutorials/MovieRecommendation/Program.cs#DataTransformations "Define data transformations")]
+
+Since `userId` and `movieId` represent users and movie titles, not real values, you use the [MapValueToKey()](xref:Microsoft.ML.ConversionsExtensionsCatalog.MapValueToKey%2A) method to transform each `userId` and each `movieId` into a numeric key type `Feature` column (a format accepted by recommendation algorithms) and add them as new dataset columns:
+
+| userId | movieId | Label | userIdEncoded | movieIdEncoded |
+| ------------- |:-------------:| -----:|-----:|-----:|
+| 1 | 1 | 4 | userKey1 | movieKey1 |
+| 1 | 3 | 4 | userKey1 | movieKey2 |
+| 1 | 6 | 4 | userKey1 | movieKey3 |
+
+
+Choose the machine learning algorithm and append it to the data transformation definitions by adding the following as the next line of code in `BuildAndTrainModel()`:
+
+[!code-csharp[AddAlgorithm](../../../samples/machine-learning/tutorials/MovieRecommendation/Program.cs#AddAlgorithm "Add the training algorithm with options")]
+
+The [MatrixFactorizationTrainer](xref:Microsoft.ML.RecommendationCatalog.RecommendationTrainers.MatrixFactorization%28Microsoft.ML.Trainers.MatrixFactorizationTrainer.Options%29) as is your recommendation training algorithm.  [Matrix Factorization](https://en.wikipedia.org/wiki/Matrix_factorization_(recommender_systems)) is a common approach to recommendation when you have data on how users have rated products in the past, which is the case for the datasets in this tutorial. There are other recommendation algorithms for when you have different data available (see the [Recommended Resources](#recommended-resources) section below to learn more). 
+
+In this case, the `Matrix Factorization` algorithm uses a method called "collaborative filtering", which assumes that if User 1 has the same opinion as User 2 on a certain issue, then User 1 is more likely to feel the same way as User 2 about a different issue.
+
+E.g. if User 1 and User 2 rate movies similarly, then User 2 is more likely to enjoy a movie that User 1 has watched and rated highly:
+
+|       | Incredibles 2 (2018) | The Avengers (2012) | Guardians of the Galaxy (2014) |
+| -------------:|-------------:| -----:|-----:|
+| User 1 | Watched and liked movie | Watched and liked movie | Watched and liked movie |
+| User 2 | Watched and liked movie | Watched and liked movie | Has not watched -- RECOMMEND movie |
+
+The `Matrix Factorization` trainer has several [Options](xref:Microsoft.ML.Trainers.MatrixFactorizationTrainer.Options) including:
```

Comment:
Can we move this section into the "Improve your model" section. We might lose some users in the explanation at this point.

Reply:
I think we can, but there will be the Options/hyperparameters visible in the code snippet, so users might wonder about that. I moved it down below and just added a "learn more below" link, let me know what you think